pkg/network/go/lutgen/run.go

```diff
@@ -0,0 +1,301 @@
+		return nil
+	}
+
+	// Store the binary struct in a list so that it can later be opened.
+	// If the command ends up failing, this will be ignored
+	// and the entire lookup table generation will exit early.
+	g.addBinary(Binary{
+		Path:            outPath,
+		GoVersion:       version,
+		GoVersionString: versionStr,
+		Architecture:    arch,
+	})
+
+	command := exec.CommandContext(
+		ctx,
+		"go",
+		"build",
+		"-o", outPath,
+		"-modfile", modFile,
+		"--",
+		g.TestProgramPath,
+	)
+
+	// Set the GOPATH and GOCACHE variables.
+	// Make sure to resolve the absolute path of install directory first.
+	installDirectoryAbs, err := filepath.Abs(g.InstallDirectory)
+	if err != nil {
+		log.Printf("error install directory at %q: %v", g.InstallDirectory, err)
+		return nil
+	}
+	command.Env = append(command.Env, fmt.Sprintf("%s=%s", "GOPATH", filepath.Join(installDirectoryAbs, "build-gopath")))
+	command.Env = append(command.Env, fmt.Sprintf("%s=%s", "GOCACHE", filepath.Join(installDirectoryAbs, "build-gocache")))
+
+	return command
+}
+
+// createFakeGoMod creates the go.mod file that the `go build` command should use
+// when compiling the test program.
+// This is needed to prevent the test program compilation from using the
+// `go.mod` file in the root of this repository.
+func (g *LookupTableGenerator) createFakeGoMod(version goversion.GoVersion, arch string) (string, error) {
+	path := filepath.Join(g.OutDirectory, fmt.Sprintf("%s.go%s.go.mod", arch, versionToString(version)))
+
+	f, err := os.Create(path)
+	if err != nil {
+		return "", fmt.Errorf("error creating file at %q: %w", path, err)
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(fmt.Sprintf("module test_program\n\ngo %d.%d\n", version.Major, version.Minor))
+	if err != nil {
+		return "", fmt.Errorf("error writing contents: %w", err)
+	}
+
+	return path, nil
+}
+
+// inspectAllBinaries runs the inspection function for each binary in parallel,
+// returning a "result table" that maps architecture,version pairs
+// to the result of the inspection.
+func (g *LookupTableGenerator) inspectAllBinaries(ctx context.Context) (map[architectureVersion]interface{}, error) {
+	// Get all of the binaries that were generated from the matrix runner
+	binaries := g.getAllBinaries()
+
+	results := make(chan struct {
+		bin    Binary
+		result interface{}
+		err    error
+	})
+	for _, bin := range binaries {
+		go func(bin Binary) {
+			result, err := g.InspectBinary(bin)
+			results <- struct {
+				bin    Binary
+				result interface{}
+				err    error
+			}{bin, result, err}
+		}(bin)
+	}
+
+	resultTable := make(map[architectureVersion]interface{})
+	for range binaries {
+		select {
+		case result := <-results:
+			if result.err != nil {
+				// Bail early and return
+				return nil, result.err
+			}
+
+			resultTable[architectureVersion{result.bin.Architecture, result.bin.GoVersion}] = result.result
+		case <-ctx.Done():
```

Comment:
where is this context getting canceled (so we wait until all binaries are inspected)? I couldn't find it 🤔

Reply:
The context gets cancelled when the lookup table generation process is stopped using SIGINT:
https://github.com/DataDog/datadog-agent/pull/10039/files#diff-58a6599cc1ee4bc326d5a57137520f5e96982bb6ec9e4cce31949c4b1b76d518R52